services/datamanager/builtin/capture_test.go

```diff
@@ -17,24 +18,39 @@ var (
 	enabledBinaryCollectorConfigPath            = "services/datamanager/data/robot_with_cam_capture.json"
 	infrequentCaptureTabularCollectorConfigPath = "services/datamanager/data/fake_robot_with_infrequent_capture.json"
 	remoteCollectorConfigPath                   = "services/datamanager/data/fake_robot_with_remote_and_data_manager.json"
-	emptyFileBytesSize = 30 // size of leading metadata message
+	emptyFileBytesSize                          = 100 // size of leading metadata message
 )
 
 func TestDataCaptureEnabled(t *testing.T) {
 	captureTime := time.Millisecond * 25
 
 	// On slower machines, it's possible that capture hasn't begun after captureTime. Give up to 20x
 	// as long for this to occur.
-	getCapturedFiles := func(captureDir string) []os.FileInfo {
-		files := getAllFiles(captureDir)
-		for i := 0; i < 20; i++ {
+	waitForCaptureFiles := func(captureDir string) {
+		files := getAllFileInfos(captureDir)
+		for i := 0; i < 50; i++ {
 			if len(files) > 0 && files[0].Size() > int64(emptyFileBytesSize) {
-				break
+				return
 			}
 			time.Sleep(captureTime)
-			files = getAllFiles(captureDir)
+			files = getAllFileInfos(captureDir)
+		}
+	}
+
+	testFilesContainSensorData := func(t *testing.T, dir string) {
+		var sd []*v1.SensorData
+		filePaths := getAllFilePaths(dir)
+		for _, path := range filePaths {
+			d, err := datacapture.SensorDataFromFilePath(path)
+			test.That(t, err, test.ShouldBeNil)
+			sd = append(sd, d...)
+		}
+
+		test.That(t, len(sd), test.ShouldBeGreaterThan, 0)
+		for _, d := range sd {
```

Comment:
I think we need to check it cumulatively, because it's still possible in the success case that some individual file actually contains no sensor data. For example, if in the time it took this to ran 65kb of data was captured, one file would have 64kb, and there would be another file with no sensor data (but 1kb of buffered sensor data waiting to be written).